pkg/controller/servicediscovery/privatednsnamespace/hooks.go

```diff
@@ -0,0 +1,160 @@
+		return ret, nil
+	}
+	opInput := &svcsdk.GetOperationInput{
+		OperationId: cr.Status.AtProvider.OperationID,
+	}
+
+	opReqResp, err := e.client.GetOperationWithContext(ctx, opInput)
+	if err != nil {
+		return ret, errors.Wrap(err, "get-operation failed")
+	}
+
+	if *opReqResp.Operation.Status == "PENDING" || *opReqResp.Operation.Status == "SUBMITTED" {
+		ret.ResourceExists = true
+		return ret, nil
+	}
+
+	if *opReqResp.Operation.Status == "FAIL" {
+		if cr.GetCondition(xpv1.TypeReady).Reason != xpv1.ReasonDeleting {
+			ret.ResourceExists = true
+		}
+		cr.Status.SetConditions(xpv1.Unavailable().WithMessage(*opReqResp.Operation.ErrorMessage))
+		return ret, nil
+	}
+
+	namespaceID, ok := opReqResp.Operation.Targets["NAMESPACE"]
+	if !ok {
+		ret.ResourceExists = true
+		return ret, errors.New(errDescribe)
+	}
+
+	nsInput := &svcsdk.GetNamespaceInput{
+		Id: namespaceID,
+	}
+	nsReqResp, err := e.client.GetNamespaceWithContext(ctx, nsInput)
+	if err != nil {
+		if ActualIsNotFound(err) || IsDuplicateRequest(err) {
+			cr.Status.SetConditions(xpv1.Unavailable())
+		}
+		return ret, awsclients.Wrap(cpresource.IgnoreAny(err, ActualIsNotFound, IsDuplicateRequest), "get-namespace failed")
+	}
+	ret.ResourceExists = true
+
+	cr.Status.SetConditions(xpv1.Available())
+	if externalName != *namespaceID {
+		meta.SetExternalName(cr, *namespaceID)
+		ret.ResourceLateInitialized = true
+	}
+
+	currentSpec := cr.Spec.ForProvider.DeepCopy()
+	lateInitialize(&cr.Spec.ForProvider, nsReqResp)
+	ret.ResourceLateInitialized = ret.ResourceLateInitialized || !cmp.Equal(&cr.Spec.ForProvider, currentSpec)
+
+	upToDate, err := e.isUpToDate(cr, nsReqResp)
+	if err != nil {
+		return ret, errors.Wrap(err, "isUpToDate check failed")
+	}
+
+	ret.ResourceUpToDate = upToDate
+	return ret, nil
+}
+
+func (e *external) updateImpl(ctx context.Context, mg cpresource.Managed) (managed.ExternalUpdate, error) {
+	// Update always returns an error, since the servicediscovery API does not allow updating namespaces
+	return managed.ExternalUpdate{}, nil
+}
+
+func (e *external) isUpToDate(cr *svcapitypes.PrivateDNSNamespace, resp *svcsdk.GetNamespaceOutput) (bool, error) { // nolint:unparam
+	if meta.GetExternalName(cr) != *resp.Namespace.Id {
+		return false, nil
+	}
+
+	if cr.Spec.ForProvider.CreatorRequestID != resp.Namespace.CreatorRequestId {
+		return false, nil
+	}
+	if cr.Spec.ForProvider.Description != resp.Namespace.Description {
+		return false, nil
+	}
+
+	// if cr.Spec.ForProvider.VPC
+	// VPC information is not available through servicediscovery API. Instead
+	// one could use the HostedZone information to verify the VPC configuration
+	// through Route53 HostedZone records
+
+	// tags := map[string]string
+	// if cmp.Equal(cr.Spec.ForProvider.Tags, tags) {
+	// Where does the servicediscovery API provide Tag information?
+
+	// if cr.Spec.ForProvider.Region
+	// Region information is not available through servicediscovery API
+	return true, nil
```

Comment:
Error in `isUpToDate` is always nil, so it can be removed.

Reply:
done